pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -0,0 +1,324 @@
+		}
+		grp.remotes[remote] = wl
+	}
+	return &grp, nil
+}
+
+func (a *wlReconciler) reconcileGroup(ctx context.Context, group *wlGroup) error {
+	log := ctrl.LoggerFrom(ctx).WithValues("op", "reconcileGroup")
+	log.V(3).Info("Reconcile Workload Group")
+
+	// 1. delete all remote workloads when finished or the local wl has no reservation
+	if group.IsFinished() || !workload.HasQuotaReservation(group.local) {
+		errs := []error{}
+		for rem := range group.remotes {
+			if err := group.RemoveRemoteObjects(ctx, rem); err != nil {
+				errs = append(errs, err)
+				log.V(2).Error(err, "Deleting remote workload", "remote", rem)
+			}
+		}
+		return errors.Join(errs...)
+	}
+
+	if remoteFinishedCond, remote := group.RemoteFinishedCondition(); remoteFinishedCond != nil {
+		//NOTE: we can have a race condition setting the wl status here and it being updated by the job controller
+		// it should not be problematic but the "From remote xxxx:" could be lost ....
+
+		if group.adaptor != nil {
+			if err := group.adaptor.CopyStatusRemoteObject(ctx, a.acr.client, group.rc.remoteClients[remote].client, group.controllerKey); err != nil {
+				log.V(2).Error(err, "copying remote controller status", "remote", remote)
+				// we should retry this
+				return err
+			}
+		} else {
+			log.V(3).Info("Group with no adaptor, skip owner status copy", "remote", remote)
+		}
+
+		// copy the status to the local one
+		wlPatch := workload.BaseSSAWorkload(group.local)
+		apimeta.SetStatusCondition(&wlPatch.Status.Conditions, metav1.Condition{
+			Type:    kueue.WorkloadFinished,
+			Status:  metav1.ConditionTrue,
+			Reason:  remoteFinishedCond.Reason,
+			Message: fmt.Sprintf("From remote %q: %s", remote, remoteFinishedCond.Message),
+		})
+		return a.acr.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName+"-finish"), client.ForceOwnership)
+	}
+
+	hasReserving, reservingRemote := group.FirstReserving()
+
+	// 1. delete all workloads that are out of sync or are not in the chosen worker
+	for rem, remWl := range group.remotes {
+		if remWl == nil {
+			continue
+		}
+		outOfSync := group.local == nil || !equality.Semantic.DeepEqual(group.local.Spec, remWl.Spec)
+		notReservingRemote := hasReserving && reservingRemote != rem
+		if outOfSync || notReservingRemote {
+			if err := group.RemoveRemoteObjects(ctx, rem); err != nil {
+				log.V(2).Error(err, "Deleting out of sync remote objects", "remote", rem)
+			}
+		}
+	}
+
+	// 2. get the first reserving
+	if hasReserving {
+		acs := workload.FindAdmissionCheck(group.local.Status.AdmissionChecks, group.acName)
+		if err := group.adaptor.CreateRemoteObject(ctx, a.acr.client, group.rc.remoteClients[reservingRemote].client, group.controllerKey, group.local.Name); err != nil {
+			log.V(2).Error(err, "creating remote controller object", "remote", reservingRemote)
+			// we should retry this
+			return err
+		}
+
+		if acs.State != kueue.CheckStateRetry {
+			acs.State = kueue.CheckStatePending
```

Comment:
this is keeping the local wl in suspended, add a comment

Reply:
The workload should be admitted on the management cluster, and suspend should be false. It is fine as is for this PR, but we should follow up. cc @mwielgus. For e2e tests we can disable the job controller on the management cluster.